modules/configuring-shared-cpus-for-a-workload.adoc

```diff
@@ -0,0 +1,106 @@
+:_mod-docs-content-type: PROCEDURE
+
+[id="configuring-shared-cpus-for-a-workload_{context}"]
+= Configuring shared CPUs in a performance profile
+
+You can define and enable shared CPUs by using a performance profile. You can use these shared CPUs for lightweight application tasks.
+
+.Prerequisites
+
+* Log in as a user with `cluster-admin` privileges.
+
+* You enabled the `TechPreviewNoUpgrade` feature set on the cluster. For more information about enabling a feature set, see the _Additional resources_ section.
++
+[WARNING]
+====
+Enabling the `TechPreviewNoUpgrade` feature set cannot be undone and prevents minor version updates. These feature sets are not recommended on production clusters.
+====
+
+.Procedure
+
+. Create a `PerformanceProfile` resource:
+
+.. Create a YAML file that defines the `PerformanceProfile` resource:
++
+.Example `shared-cpu-pp.yaml` file
+[source,yaml]
+----
+apiVersion: performance.openshift.io/v2
+kind: PerformanceProfile
+metadata:
+  name: example-shared-cpu-pp
+spec:
+  cpu:
+    reserved: "0-1"
+    isolated: "4-8"
+    shared: "2-3" <1>
+  workloadHints:
+    mixedCpus: true <2>
+  nodeSelector:
+    node-role.kubernetes.io/performance: "test" <3>
+----
+<1> Define the shared CPU cores. The number of shared cores must be less than the number of reserved cores.
+<2> Enable the shared CPUs feature by enabling the `mixedCpus` workload hint.
+<3> Select the node that you want to use for shared CPUs.
+
+.. Create the `PerformanceProfile` resource by running the following command:
++
+[source,bash]
+----
+$ oc create -f shared-cpu-pp.yaml
+----
++
+.Example output
+[source,terminal]
+----
+performanceprofile.performance.openshift.io/example-shared-cpu-pp created
+----
+
+
+.Verification
+
+. Verify that the performance profile is created by running the following command:
++
+[source,terminal]
+----
+$ oc describe performanceprofile example-shared-cpu-pp
+----
++
+.Example output
+[source,terminal]
+----
+...
+Events:
+  Type    Reason              Age                 From                            Message
+  ----    ------              ----                ----                            -------
+  Normal  Creation succeeded  27m (x17 over 35m)  performance-profile-controller  Succeeded to create all components
+----
+
+. Verify that the shared CPUs are present under the `reservedSystemCPUs` pool:
+
+.. Enter into a debug session on the target node:
++
+[source,terminal]
+----
+$ oc debug node/<node_name>
+----
+
+.. Set `/host` as the root directory within the debug shell prompt.
++
+[source,terminal]
+----
+sh-4.4# chroot /host
+----
+
+.. Check that the `reservedSystemCPUs` pool includes the `shared` and `reserved` CPUs by running the following command:
```

Comment:
@Tal-or do we know what this looks like when shared CPUs are enabled and defined? I'm just guessing as I don't have access to a configured environment. Is this the best way to verify the shared CPUS? Thanks for any help.

Reply:
This is one way to achieve that. I would only specify that the command should be running under the node.

Reply:
I think the first sub-step mentions that is should be done on the target node.
`Enter into a debug session on the target node:` 
I updated the first step too to make it clear that these verification steps apply only to the target node:
`Verify that the shared CPUs are present under the `reservedSystemCPUs` pool on the target node:`